src/Signature/SignatureProvider.php

```diff
@@ -114,6 +114,8 @@ public static function version()
                     return $service === 's3'
                         ? new S3SignatureV4($service, $region)
                         : new SignatureV4($service, $region);
+                case 'v4-unsigned-body':
+                    return new SignatureV4($service, $region, $options = ['unsigned' => 'true']);
```

Comment:
The `'unsigned'` parameter name sound a bit vague, is might cause the confusion that it's trying to unsign request completely. Could you rename it perhaps something like `unsign-body`?